### PR TITLE
Move compute_ctl structs used in HTTP API and spec file to separate crate.

### PR DESCRIPTION
**Pull request by hlinnaka**
This is in preparation of using compute_ctl to launch postgres nodes in the neon_local control plane. And seems like a good idea to separate the public interfaces anyway.

One non-mechanical change here is that we now use a RwLock rather than atomics to protect the ComputeNode::metrics field. We were not using atomics for performance but for convenience here, and an RwLock is now more convenient.

(extracted to separate PR from PR #3886)

### CODE REVIEW

**Review by ololobus — COMMENTED** [review]
Moving structures in a separate crate sounds as good first step. Also moving metrics under the same lock should be fine too, as actually they are set now only during initial startup, so it's basically the same writer all the time

Yet, I made state to be under `Mutex` in the https://github.com/neondatabase/neon/pull/3923 because of two reasons:
1. Condvar was super convenient for my use case
2. Write lock in the `RwLock` anyway blocks all readers, IIUC. And we don't have a case when there are a lot of concurrent readers

What do you think?

We should also agree on the order of PRs, probably :) Yours seems to be important to write better tests in the `neon` repo. Mine is needed to unblock control-plane work on reconfiguration without restart and maintaining the pool of pre-created computes. So not sure, just your refactoring looks relatively huge in general (if we consider all needed PRs)

**Comment by hlinnaka** [comment]
> Moving structures in a separate crate sounds as good first step. Also moving metrics under the same lock should be fine too, as actually they are set now only during initial startup, so it's basically the same writer all the time
> 
> Yet, I made state to be under `Mutex` in the #3923 because of two reasons:
> 
>     1. Condvar was super convenient for my use case
> 
>     2. Write lock in the `RwLock` anyway blocks all readers, IIUC. And we don't have a case when there are a lot of concurrent readers
> 
> 
> What do you think?

Sure, Mutex is fine.

> We should also agree on the order of PRs, probably :) Yours seems to be important to write better tests in the `neon` repo. Mine is needed to unblock control-plane work on reconfiguration without restart and maintaining the pool of pre-created computes. So not sure, just your refactoring looks relatively huge in general (if we consider all needed PRs)

First to push wins ;-).

I suggest:
1. #3937
2. #3938
3. #3923 

I can rebase #3923 over those two, I think the conflicts are pretty mechanical

**Comment by hlinnaka** [comment]
Rebased, fixing conflicts with commit e42982fb1e0178f5efe338a20e5fd8e593aa4ffb.

I think this is ready to be merged. Please review and approve if you agree.

File: libs/compute_api/src/models.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+/// Request of the /configure API
+///
+/// We now pass only `spec` in the configuration request, but later we can
+/// extend it and something like `restart: bool` or something else. So put
+/// `spec` into a struct initially to be more flexible in the future.
+#[derive(Deserialize, Debug)]
+pub struct ConfigurationRequest {
+    pub spec: ComputeSpec,
+}
```
Comment (ololobus) [inline libs/compute_api/src/models.rs:15]: What do you think about separating http request/responses into different files as it's in the main now? It makes usage and navigation more clean

We can keep common parts in the `models.rs`, though

**Review by ololobus — APPROVED** [review]
LGTM, one comment / suggestion about files layout and this text

> One non-mechanical change here is that we now use a RwLock rather than atomics to protect the ComputeNode::metrics field. We were not using atomics for performance but for convenience here, and an RwLock is now more convenient.

should be updated as we use Mutex now.

File: libs/compute_api/src/models.rs (hunk 1 above)
Comment (hlinnaka) [reply libs/compute_api/src/models.rs]: Ok, I split it up again. There's nothing but ConfigurationRequest in requests.rs now, but maybe we'll grow more.

**Pull request merged**